File: 118.pascals-triangle.cpp
/*
 * @lc app=leetcode id=118 lang=cpp
 *
 * [118] Pascal's Triangle
 */
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> g;
        vector<int> temp;
        if(numRows == 1)
        {
            temp.push_back(1);
            g.push_back(temp);
            return g;
        }
       
        for(int i = 0; i < numRows; ++i)
        {
            temp.push_back(1);//在每一层的开始的地方弹入一个1
            for(int j = 1; j < i; ++j)
            {
                temp.push_back(g[i - 1][j] + g[i - 1][j -1]);
            }
            if(i > 0)
            temp.push_back(1);//在每层的结束地方弹入一个1
            g.push_back(temp);
            temp.clear();
        }
        return g;
    }
};
/* 杨辉三角 首先要保证第一层是1
后面每一层的第一个都是1 最后一个都是1

File: 118.pascals-triangle.py
#
# @lc app=leetcode id=118 lang=python3
#
# [118] Pascal's Triangle
#
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        result = []
        for i in range(numRows):
            now = [1]*(i+1)
            if i >= 2:
                for n in range(1,i):
                    now[n] = pre[n-1]+pre[n]
            result += [now]
            pre = now
        return result

'''先让所有的now的值都为1 这就免除了专门去定义开头和末尾值的过程
之后的做法和c++类似但是要注意两者语法的区别，注意python中for循环
范围的限定